### PR TITLE
Support input type email

### LINKED ISSUE #1444: Use <input type=email> if value is email

### LINKED ISSUE #1449: Render labels exactly according to configuration
- [ ] `Use email or username instead` -> Look at configuration and use the exact wordings.

### PR DESCRIPTION
**Pull request by louischan-oursky**
fix #1444
fix #1449

### CODE REVIEW

**Comment by louischan-oursky** [comment]
But the login field is supposed to be shared, i.e. the user could enter anything (including phone number) We cannot know in advance the user wants to input an email or a username.

**Comment by carmenlau** [comment]
I mean if the app config supported login id have email and username, we should use text. And the login id has email only, we should use email. what do you think?

**Comment by louischan-oursky** [comment]
Updated.

File: pkg/auth/dependency/webapp/template.go [hunk 1]
```diff
@@ -304,14 +304,14 @@ var TemplateAuthUILoginHTML = template.Spec{
...
+				<input class="input text-input primary-txt" type="{{ .x_login_page_login_id_input_type }}" name="x_login_id" placeholder="{{ localize "login-id-placeholder" }}" value="{{ .x_login_id }}">
 				{{ end }}{{ end }}
 
-				{{ if .x_login_id_input_type }}{{ if and (eq .x_login_id_input_type "phone") .x_login_id_input_type_has_text }}
-				<a class="link align-self-flex-start" href="{{ call .MakeURLWithQuery "x_login_id_input_type" "text" }}">{{ localize "use-text-login-id-description" }}</a>
+				{{ if .x_login_id_input_type }}{{ if eq .x_login_id_input_type "phone" }}
+				<a class="link align-self-flex-start" href="{{ call .MakeURLWithQuery "x_login_id_input_type" .x_login_page_login_id_input_type }}">{{ localize "use-text-login-id-description" }}</a>
 				{{ end }}{{ end }}
-				{{ if .x_login_id_input_type }}{{ if and (not (eq .x_login_id_input_type "phone")) .x_login_id_input_type_has_phone }}
+				{{ if .x_login_id_input_type }}{{ if (not (eq .x_login_id_input_type "phone")) }}
```
Comment (carmenlau) [inline pkg/auth/dependency/webapp/template.go:314]: After updating, the link `Use phone number instead` is shown if I enable `username` and `email` only.

File: pkg/auth/dependency/webapp/translation.go [hunk 2]
```diff
@@ -43,8 +43,8 @@ var TemplateAuthUITranslationJSON = template.Spec{
 	"sso-login-id-separator": "or",
 
 	"phone-number-placeholder": "Phone number",
-	"login-id-placeholder": "Email or username",
-	"use-text-login-id-description": "Use email or username instead",
+	"login-id-placeholder": "{0, select, email {email} username {username} other {email or username}}",
```
Comment (carmenlau) [inline pkg/auth/dependency/webapp/translation.go:46]: Across the whole auth ui, the first character of the placeholder should be upper case. Please update it for consistency.

Comment (louischan-oursky) [reply pkg/auth/dependency/webapp/translation.go]: Fixed.

**Pull request merged**